### DISCUSSION TITLE
Why do a polyglot runtime?

### DISCUSSION THREAD

**Discussion by Olyno** [oracle/graal#5068] [General] [answered]
Hi 👋

I've been wondering for a while, but what are the motivations to create a polyglot runtime, able to execute JavaScript, Java, R... ?

As far as I know, Graal doesn't bring anything new on the JS part, and the JS world already has 3 big runtime names, namely Node, Deno and Bun.

No criticism, I'm just trying to understand the motivations of the project. If you have documentation, interviews, or any other link that could answer my question, I would be grateful!

**Comment by smarr** [comment, marked as answer]
One aspect is using existing libraries from other languages, various demos here: https://github.com/hpi-swa/trufflesqueak#demos
The polyglot approach and the language interoperability approach also has the benefit that one can use a single mechanism from each language, instead of relying on language-specific solutions.


Another is that one can use the various tools for different languages, which avoids having to learn different tools for each language.

**Reply by Olyno** [reply]
I see, thank you very much!
